Seda/Seda/Controllers/PaymentViewController.swift

```diff
@@ -0,0 +1,157 @@
+            displayNumber += String(char)
+        }
+        return displayNumber
+    }
+    
+    func format16(number: String) -> String {
+        var displayNumber = ""
+        for (idx, char) in number.enumerated() {
+            if (idx % 4) == 0 && idx != 0 {
+                displayNumber += " "
+            }
+            displayNumber += String(char)
+        }
+        return displayNumber
+    }
+    
+    func startCheckout() {
+        // Request a PaymentIntent from your server and store its client secret
+        // Create a PaymentIntent by calling the sample server's /create-payment-intent endpoint.
+        
+        let url = URL(string: backendUrl + "create-payment-intent")!
+        let json: [String: Any] = [
+          "items": [
+              ["id": "CASHTOSEND"]//TODO: CASHTOSEND should be taken from the UITextField, checked that it's a valid amount, and sent to our server
+          ]
+        ]
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.httpBody = try? JSONSerialization.data(withJSONObject: json)
+        let task = URLSession.shared.dataTask(with: request, completionHandler: { [weak self] (data, response, error) in
+          guard let response = response as? HTTPURLResponse,
+            response.statusCode == 200,
+            let data = data,
+            let json = try? JSONSerialization.jsonObject(with: data, options: []) as? [String : Any],
+            let clientSecret = json["clientSecret"] as? String else {
+                let message = error?.localizedDescription ?? "Failed to decode response from server."
+                self?.displayAlert(title: "Error loading page", message: message)
+                return
+          }
+          print("Created PaymentIntent")
+          self?.paymentIntentClientSecret = clientSecret
+            print("Set Client Secret")
+            // Configure the SDK with your Stripe publishable key so that it can make requests to the Stripe API
+            // For added security, our sample app gets the publishable key from the server
+            //Stripe.setDefaultPublishableKey(publishableKey)
+        })
+        task.resume()
+    }
+    
+    func displayAlert(title: String, message: String, restartDemo: Bool = false) {
+        DispatchQueue.main.async {
+            let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
+            if restartDemo {
+                alert.addAction(UIAlertAction(title: "Restart demo", style: .cancel) { _ in
+                    self.cardTextField.clear()
+                    self.startCheckout()
+                })
+            }
+            else {
+                alert.addAction(UIAlertAction(title: "OK", style: .cancel))
+            }
+            self.present(alert, animated: true, completion: nil)
+        }
+    }
+    
+    @objc
+    func pay() {
```

Comment:
Is this handling peer to peer pay or is this loading a balance onto the user's account?

Reply:
This is for loading money onto a user account. It still needs to be implemented to update a user's cash balance though, but that'll be done after firebase is setup with stripe.